src/renderers/dom/client/wrappers/ReactDOMTextarea.js

```diff
@@ -66,12 +65,14 @@ var ReactDOMTextarea = {
       '`dangerouslySetInnerHTML` does not make sense on <textarea>.'
     );
 
-    // Always set children to the same thing. In IE9, the selection range will
-    // get reset if `textContent` is mutated.
```

Comment:
The issue is that no matter how you might update the children, IE loses cursor position in textareas only. Elements with contenteditable should be handled by our ReactInputSelection wrapper on the transaction. I'd like this logic to be maintained as it was, where the children are always passed down the same way.